### ISSUE TITLE
Improve Link UI anchor position in Navigation block submenus

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#68182]
## What problem does this address?

When creating menu items within submenus within the Navigation block the Link UI popover will remain open after the link is created. This is intentional in order to allow users to then adjust the settings of the link (e.g. `Open in new tab`).

However, the popover can obscure the `+` button inserter.

## What is your proposed solution?

Either

- improve the anchor positioning of the Link UI so that it takes into account the submenu (note there was [work to do the same for the block toolbar recently](https://github.com/WordPress/gutenberg/pull/62711)).
- hide the Link UI following creation - note however that this would require another click to modify any link settings. User feedback was that this was suboptimal so we should be careful before changing this behaviour.

**Comment by snehapatil2001** [comment]
Hey @getdave, Thank you for raising this issue! 

- I’ve tested the current behaviour, and I’d like to share a video demonstration to show that after clicking on a submenu item, the popover disappears, and the + button becomes visible again. 

- Given this behaviour, it seems that no changes are necessary to address this issue.

https://github.com/user-attachments/assets/04af2fb6-a00c-44bf-9689-8947ad9b528a

Hey @yogeshbhutkar 

- I’ve reviewed your implementation, and I wanted to share some feedback based on my testing. If there are multiple submenus, the current margin-top logic does not work consistently. 

- For example, when opening the popover on a submenu item, it can still obscure the + button of other sub-items. Please see the attached video:

https://github.com/user-attachments/assets/9693aaf4-cd2a-4a4c-9997-dd7239f624e5

**Comment by yogeshbhutkar** [comment]
Yes @snehapatil2001, I believe you're right here. I kind of only took into consideration the last element. But, that shouldn't be the case. I think we'll need to update the logic of calculating the positions a bit. I’m working on it now and will share an update once it’s ready.

Thanks.
